components/lvgl_esp32_drivers/lvgl_driver.c

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file lvgl_driver.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "sdkconfig.h"
+#include "lvgl_driver.h"
+
+#include "lvgl_tft/disp_spi.h"
+#include "lvgl_touch/tp_spi.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+void lvgl_driver_init(void)
+{
+    /* Interface and driver initialization */
+#ifdef SHARED_SPI_BUS
+	/* Configure one SPI bus for the two devices */
+	configure_shared_spi_bus();
+    
+	/* Configure the drivers */
+	disp_driver_init(false);
+#if CONFIG_LVGL_TOUCH_CONTROLLER != TOUCH_CONTROLLER_NONE
+	touch_driver_init(false);
+#endif
+#else
+	/* Otherwise configure the SPI bus and devices separately inside the drivers*/
+	disp_driver_init(true);
+#if CONFIG_LVGL_TOUCH_CONTROLLER != TOUCH_CONTROLLER_NONE
+	touch_driver_init(true);
+#endif
+#endif  // SHARED_SPI_BUS
+}
+
+void configure_shared_spi_bus(void)
```

Comment:
Can we make this function static?

Reply:
Sure, it is done.

Reply:
I get the `unused function` warning when no using a touch controller, do you think we can do something like:
```c
#if CONFIG_LVGL_TOUCH_CONTROLLER != TOUCH_CONTROLLER_NONE
static void configure_shared_spi_bus(void);
#endif

...

#if CONFIG_LVGL_TOUCH_CONTROLLER != TOUCH_CONTROLLER_NONE
static void configure_shared_spi_bus(void)
{
    /* implementation */
}
#endif
```

What do you think?